src/mongo/shell/encrypted_dbclient_base.cpp

```diff
@@ -677,10 +675,25 @@ std::shared_ptr<SymmetricKey> EncryptedDBClientBase::getDataKeyFromDisk(const UU
         _encryptionOptions.getKmsProviders().toBSON(), keyStoreRecord.getMasterKey());
     SecureVector<uint8_t> decryptedKey =
         kmsService->decrypt(dataKey, keyStoreRecord.getMasterKey());
+    return decryptedKey;
+}
+
+std::shared_ptr<SymmetricKey> EncryptedDBClientBase::getDataKeyFromDisk(const UUID& uuid) {
+    auto decryptedKey = getKeyMaterialFromDisk(uuid);
     return std::make_shared<SymmetricKey>(
         std::move(decryptedKey), crypto::aesAlgorithm, "kms_encryption");
 }
 
+KeyMaterial EncryptedDBClientBase::getKey(const UUID& uuid) {
```

Comment:
Just realized `KeyMaterial` is now a `SecureVector<uint8_t>`, so this is not needed. Can just call `getKeyMaterialFromDisk()' directly.

Reply:
oh nvm. Size needs to be truncated to 32 bytes.. so this is still needed for FLE2. Although, the std::copy below takes the sizeof(KeyMaterial), which doesn't seem right..

Reply:
oh nvm. Size needs to be truncated to 32 bytes.. so this is still needed for FLE2. Although, the std::copy below takes the sizeof(KeyMaterial), which doesn't seem right..

Reply:
Yeah, this was wrong after the rebase onto SecureVector. It failed during my testing as I merged this. I changed it to
```
    KeyMaterial km;
    km->resize(decryptedKey->size());
    std::copy(decryptedKey->data(), decryptedKey->data() + decryptedKey->size(), km->data());```